src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat.pop();
+  const separatorTo = toFormat.pop();
+  const dateArr = date.split(separatorFrom);
+  const resultArr = [];
+
+  for (let i = 0; i < toFormat.length; i++) {
+    const format = toFormat[i];
+    const index = fromFormat.indexOf(format);
+
+    if (index === -1) {
+      if (format === 'YYYY') {
+        const fromIndex = fromFormat.indexOf('YY');
+        const year = dateArr[fromIndex];
+
+        resultArr.push(year < 30 ? '20' + year : '19' + year);
+      } else if (format === 'YY') {
+        const fromIndex = fromFormat.indexOf('YYYY');
+        const year = dateArr[fromIndex];
+
+        resultArr.push(year.slice(-2));
+      }
```

Comment:
The logic for converting the year does not handle the case where the corresponding year format (`YYYY` or `YY`) is not found in `fromFormat`. This could result in `fromIndex` being `-1` and consequently `undefined` being used in `dateArr[fromIndex]`, which would lead to incorrect results.

Reply:
Thank you, this will be useful in the future. In this task, the date formats are given.